### PR TITLE
Add code coverage, tests of initializing and remove not used code

### PR DESCRIPTION
**Pull request by Semigradsky**
Code coverage
===

Before:
--
> 97.12% Statements 2093/2155
> 92.69% Branches 1192/1286
> 97.45% Functions 382/392
> 97.23% Lines 2069/2128

After:
--
> 97.67% Statements 2099/2149
> 93.23% Branches 1199/1286
> 97.95% Functions 382/390
> 97.79% Lines 2075/2122

### CODE REVIEW

File: lib/prefixes.js [hunk 1]
```diff
@@ -51,7 +51,6 @@ Value.hack(require('./hacks/intrinsic'));
 Value.hack(require('./hacks/pixelated'));
 Value.hack(require('./hacks/image-set'));
 Value.hack(require('./hacks/cross-fade'));
-Value.hack(require('./hacks/flex-values'));
```
Comment (ai) [inline lib/prefixes.js]: Are you sure about this changes? Looks dangerous.

File: lib/autoprefixer.js [hunk 2]
```diff
@@ -28,40 +28,52 @@ function timeCapsule(result, prefixes) {
     );
 }
 
-module.exports = postcss.plugin('autoprefixer', (...reqs) => {
+function getOptions(...args) {
```
Comment (ai) [inline lib/autoprefixer.js]: Why not just `getOptions(args)`? Faster and simpler ;).

File: lib/autoprefixer.js [hunk 3]
```diff
@@ -28,40 +28,52 @@ function timeCapsule(result, prefixes) {
...
-    } else if (reqs.length === 0 || reqs.length === 1 && !reqs[0]) {
-        reqs = undefined;
-    } else if (reqs.length <= 2 && (reqs[0] instanceof Array || !reqs[0])) {
-        options = reqs[1];
-        reqs = reqs[0];
-    } else if (typeof reqs[reqs.length - 1] === 'object') {
-        options = reqs.pop();
+    let reqs;
+
+    // autoprefixer([...,] options)
```
Comment (ai) [inline lib/autoprefixer.js]: I prefer to not write this type of comments. If you need comment, you should make code clearly.

File: lib/prefixes.js (hunk 1 above)
Comment (ai) [reply lib/prefixes.js]: Yeap, it is dangerous. We use it for `transition: flex-grow`.

Comment (Semigradsky) [reply lib/prefixes.js]: Oh I see... I have returned file.
Would be good to add test for this case.

File: lib/autoprefixer.js (hunk 2 above)
Comment (Semigradsky) [reply lib/autoprefixer.js]: Exactly!

File: lib/autoprefixer.js (hunk 3 above)
Comment (Semigradsky) [reply lib/autoprefixer.js]: I have rewrote a little.

**Comment by ai** [comment]
I don’t like many changes in the code. We had bad story in Browserslist when coverage fixes broke release.

What do you think of not changing origin code and only add new tests?

**Comment by Semigradsky** [comment]
Ok, I will change PR. I will ping when it be done.

**Comment by Semigradsky** [comment]
@ai ping! :point_left:

File: test/autoprefixer.test.js [hunk 4]
```diff
@@ -145,9 +145,33 @@ it('throws on wrong options', () => {
     }).toThrowError(/browsers/);
 });
 
-it('sets options', () => {
-    const opts = { browsers: ['chrome 25', 'opera 12'], cascade: false };
-    expect(autoprefixer(opts).options).toEqual(opts);
+describe('sets options', () => {
```
Comment (ai) [inline test/autoprefixer.test.js]: You don’t need `description` here. Also it double the test name

File: test/autoprefixer.test.js [hunk 5]
```diff
@@ -145,9 +145,33 @@ it('throws on wrong options', () => {
...
+    const opts = {
+        browsers: ['chrome 25', 'opera 12'],
+        cascade: false,
+        grid: false
+    };
+
+    it('sets options (1)', () => {
```
Comment (ai) [inline test/autoprefixer.test.js]: `1, 2, 3` is bad tests names

File: lib/autoprefixer.js [hunk 6]
```diff
@@ -58,6 +58,8 @@ module.exports = postcss.plugin('autoprefixer', (...reqs) => {
 
     if (options.browsers) {
         reqs = options.browsers;
+    } else {
```
Comment (ai) [inline lib/autoprefixer.js]: Why we need this change?

Comment (Semigradsky) [reply lib/autoprefixer.js]: If we want have `browsers` in `options`.

Comment (ai) [reply lib/autoprefixer.js]: But we don’t want to change API ;)

Comment (Semigradsky) [reply lib/autoprefixer.js]: I'm not sure how check setting `browsers` without getting it from `options` after initialize.

Comment (ai) [reply lib/autoprefixer.js]: Why you even need it? You improve tests, yeap? Not a API?

Comment (ai) [reply lib/autoprefixer.js]: Ouh, you mean that you need to get target browsers for tests? Changing original `options`  object is a bad way to do it.

You can use `info()` or check output prefixes.

Comment (Semigradsky) [reply lib/autoprefixer.js]: So weird. We have specified the browsers but can't get him by `instance.options`

Comment (ai) [reply lib/autoprefixer.js]: Or you can add special `plugin.browsers` with browsers after `browserslist`

File: test/autoprefixer.test.js [hunk 7]
```diff
@@ -145,9 +145,33 @@ it('throws on wrong options', () => {
     }).toThrowError(/browsers/);
 });
 
-it('sets options', () => {
-    const opts = { browsers: ['chrome 25', 'opera 12'], cascade: false };
-    expect(autoprefixer(opts).options).toEqual(opts);
+describe('sets options via', () => {
```
Comment (ai) [inline test/autoprefixer.test.js]: We don’t use this type of grouping in other tests. It will be better to remove `describe` in terms of tests consistency

**Comment by Semigradsky** [comment]
Added `plugin.browsers`

File: lib/autoprefixer.js [hunk 8]
```diff
@@ -92,6 +92,8 @@ module.exports = postcss.plugin('autoprefixer', (...reqs) => {
 
     plugin.options = options;
 
+    plugin.browsers = reqs;
```
Comment (ai) [inline lib/autoprefixer.js]: It is be better to put not a `reqs` here, but browsers after Browserslist ;)

**Pull request merged**